common/tests/functional/test_cbma/cbma_mdm_integration.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/sh
+
+
+MDM_AGENT_DIR="/opt/mesh_com/modules/sc-mesh-secure-deployment/src/nats"
+MDM_AGENT_CERTS="${MDM_AGENT_DIR}/certs"
+
+MDM_SERVER_DIR="~/mdm-server"
+MDM_SERVER_BIND_IP="0.0.0.0"
+MDM_SERVER_PORT="5000"
+
+LAUNCH_MDM_SERVER=0
+DEBUG=0
+
+
+###########################################################################
+
+set -e
+
+MDM_SERVER_IP="$1"
+shift
+
+MDM_AGENT_IPS="$@"
+
+
+[ "$DEBUG" != "1" ] || unset DEBUG
+
+
+usage() {
+    echo "Usage: $0 <mdm-server-ip> <csl1-ip> [<csl2-ip> ...]" >&2
+    exit 1
+}
+
+setup_ssh() {
+    ip="$1"
+
+    echo -n "[+] Setting up SSH setup for ${ip}... "
+
+    command ssh "root@${ip}" "cat ~/.ssh/id_rsa" | ssh-add - >/dev/null 2>&1
```

Comment:
Not sure I understand this correctly: we are connecting to the remote host (first manually typing the password I guess) to cat their private key back into the local SSH agent? 🤔 I guess this technically works but only as long as the CSLs have that RSA key pair built in, which Govind promised is only in debug versions and which eventually needs to be replaced with certificates anyway... So for security reasons and especially if the remote .ssh/id_rsa file might not exist, what should be done instead is to add a local public key (the .ssh/id_*.pub files) to the remote .ssh/authorized_keys, which can be done with 'cat >>' of course but nowadays there is even the ssh-copy-id command to facilitate this 🙂

Reply:
Done!

It is missing the cleanup of the added key in `.ssh/authorized_keys`, maybe is not a big deal but let me know if you have concerns with having leftover keys there :dagger: